libs/pipeline.py

```diff
@@ -0,0 +1,89 @@
+"""
+Code that is used to help move information around in the pipeline, starting with `Region` which
+represents a geographical area (state, county, metro area, etc).
+"""
+
+
+import json
+import os
+from typing import Optional
+
+import pandas as pd
+from pydantic import BaseModel
+
+import pyseir
+from covidactnow.datapublic.common_fields import CommonFields
+from libs.datasets import FIPSPopulation
+from libs.datasets.combined_datasets import load_us_latest_dataset, _log
+from pyseir.rt.utils import NEW_ORLEANS_FIPS
+from pyseir.utils import get_run_artifact_path, RunArtifact
+
+
+class Region(BaseModel):
+    """Identifies and provides access to data about a geographical area."""
+
+    # The FIPS identifier for the region, either 2 digits for a state or 5 digits for a county.
+    fips: str
+
+    def __repr__(self) -> str:
+        return f"Region(fips={self.fips})"
+
+    def get_us_latest(self):
+        """Gets latest values for a given state or county fips code."""
+        us_latest = load_us_latest_dataset()
+        return us_latest.get_record_for_fips(self.fips)
+
+    def load_inference_result(self):
```

Comment:
I would push back against having `load_{inference_result,rt_result,ensemble_result}` in this class.  Those output files are pure intermediaries and I think it muddies what this Region object represents a bit.  It feels more natural to say "i'd like the population for this region" - less so to say "i'd like the output of a previous step in the pyseir code".

Reply:
having a module like `pyseir_pipeline_intermediaries` where `load_inference_result` could live feels a bit more clear.

Reply:
I created three classes and while they could be made related by inheritance I don't think that helps us so I didn't do it. Now `Region` is a pure identifier and there are other classes for passing around data with a `Region`. Good enough to ship? @ghop02